### ISSUE TITLE
Unexpected error

### ISSUE THREAD

**Issue by nadilas** [FusionAuth/fusionauth-issues#217]
Hello,

I haven't logged in during the past couple of weeks into the system. 
Now I get the following error message on login:
`A request to the search index has failed. This error is unexpected. Contact Support.`

Both fusionauth and fusionauth-search are running as windows services connected to a local postgresql.

Thanks

**Comment by robotdan** [comment]
Is there anything in the logs? `/usr/local/fusionauth/logs/fusionauth-app.log` or `/usr/local/fusionauth/logs/fusionauth-search.log`

**Comment by nadilas** [comment]
Hi @robotdan, nothing particularly interesting or related to this message, just the startup. I tried to let the search settle first (~5 minutes) and then started fusionauth. Just to note: it was working before.

**Comment by robotdan** [comment]
Are you able to recover by restarting search? 

Whenever we throw that exception we will log an error to the FusionAuth log.

**Comment by nadilas** [comment]
Recovering by restarting search doesn't work.
The last log entries from fusionauth-search are from Tuesday. Same for fusionauth (stderr, stdout) with the one exception: 
`Jul 11, 2019 2:56:14.107 PM INFO  io.fusionauth.api.service.search.ElasticSearchClientProvider - Connecting to FusionAuth Search Engine at [http://localhost:9021]`

**Comment by pitcons** [comment]
Hello, I have the same issue with docker.

The message on login: "A request to the search index has failed. This error is unexpected. Contact Support."

Symptoms: Space for Elasticsearch was low: "search_1      | [2019-07-28T08:38:33,345][WARN ][o.e.c.r.a.DiskThresholdMonitor] [UuDGgtF] high disk watermark [90%] exceeded on [UuDGgtFxSI2Bqevlo6rcJg][UuDGgtF][/usr/share/elasticsearch/data/nodes/0] free: 9.2gb[5%], shards will be relocated away from this node"

I made more free space.

Elasticsearch health:
```
curl -XGET 'localhost:9200/_cluster/health?pretty'
{
  "cluster_name" : "fusionauth",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 5,
  "active_shards" : 5,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 5,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 50.0
}

```

Fusinauth logs:
```
fusionauth_1  | Jul 28, 2019 10:39:41.042 AM INFO  com.inversoft.scheduler.DefaultScheduler - Starting up scheduler
fusionauth_1  | Jul 28, 2019 10:39:41.043 AM INFO  com.inversoft.scheduler.DefaultScheduler - Scheduler is running
fusionauth_1  | Jul 28, 2019 10:39:41.065 AM INFO  io.fusionauth.api.service.system.NodeService - Add node with address [http://fusionauth:9011]
fusionauth_1  | Jul 28, 2019 10:39:41.421 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_2_0]
fusionauth_1  | Jul 28, 2019 10:39:41.427 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_3_0]
fusionauth_1  | Jul 28, 2019 10:39:41.438 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_3_1]
fusionauth_1  | Jul 28, 2019 10:39:41.451 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_5_0]
fusionauth_1  | Jul 28, 2019 10:39:41.475 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_6_0]
fusionauth_1  | Jul 28, 2019 10:39:41.481 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_7_0]
fusionauth_1  | 28-Jul-2019 10:39:41.668 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-9011"]
fusionauth_1  | 28-Jul-2019 10:39:41.690 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["https-jsse-nio-9013"]
fusionauth_1  | 28-Jul-2019 10:39:41.732 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["ajp-nio-9019"]
fusionauth_1  | 28-Jul-2019 10:39:41.826 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in 17933 ms
fusionauth_1  | Jul 28, 2019 10:39:48.776 AM INFO  io.fusionauth.api.service.search.ElasticSearchClientProvider - Connecting to FusionAuth Search Engine at [http://search:9200]
fusionauth_1  | Jul 28, 2019 10:40:18.419 AM INFO  io.fusionauth.api.service.system.NodeService - Remove node [http://fusionauth:9011] because it has not checked in for the last 89 seconds.
fusionauth_1  | Jul 28, 2019 10:42:34.409 AM INFO  io.fusionauth.api.service.system.NodeService - Became the master Node with Id [d81201d2-40c3-48ff-b153-2aa68f070386] at timestamp [2019-07-28T10:42:34.409Z]. Took over from [a70479be-7d4e-4cd3-a2e0-c7d3d6e57a72] who last updated at [2019-07-28T10:35:33.590Z]
fusionauth_1  | 28-Jul-2019 10:44:28.666 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: [/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib]
fusionauth_1  | 28-Jul-2019 10:44:29.396 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-9011"]
fusionauth_1  | 28-Jul-2019 10:44:29.443 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
fusionauth_1  | 28-Jul-2019 10:44:29.461 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["https-jsse-nio-9013"]
fusionauth_1  | 28-Jul-2019 10:44:30.794 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
fusionauth_1  | 28-Jul-2019 10:44:30.832 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["ajp-nio-9019"]
fusionauth_1  | 28-Jul-2019 10:44:30.847 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
fusionauth_1  | 28-Jul-2019 10:44:30.864 INFO [main] org.apache.catalina.startup.Catalina.load Initialization processed in 3470 ms
fusionauth_1  | 28-Jul-2019 10:44:30.883 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
fusionauth_1  | 28-Jul-2019 10:44:30.884 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet Engine: Apache Tomcat/8.5.31


fusionauth_1  | Jul 28, 2019 10:44:36 AM org.apache.jasper.servlet.TldScanner scanJars
fusionauth_1  | INFO: At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
fusionauth_1  | Jul 28, 2019 10:44:40.824 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - Loading FusionAuth configuration file [/usr/local/fusionauth/fusionauth-app/apache-tomcat/../../config/fusionauth.properties]
fusionauth_1  | Jul 28, 2019 10:44:40.831 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - FusionAuth Runtime Mode [Production]
fusionauth_1  | Jul 28, 2019 10:44:40.831 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - Override property [fusionauth-app.url] set to [http://fusionauth:9011] using FUSIONAUTH_URL environment value.
fusionauth_1  | Jul 28, 2019 10:44:41.125 AM INFO  com.inversoft.maintenance.DefaultMaintenanceModeSilentConfigurationWorkflow - 
fusionauth_1  | 
fusionauth_1  | ---------------------------------------------------------------------------------------------------------
fusionauth_1  | ---------------------------------- Entering Silent Configuration Mode -----------------------------------
fusionauth_1  | ---------------------------------------------------------------------------------------------------------
fusionauth_1  | 
db_1          | LOG:  incomplete startup packet
fusionauth_1  | Jul 28, 2019 10:44:41.644 AM INFO  com.inversoft.maintenance.db.JDBCMaintenanceModeDatabaseService - Database Version [1.7.2]
fusionauth_1  | Jul 28, 2019 10:44:41.644 AM INFO  com.inversoft.maintenance.db.JDBCMaintenanceModeDatabaseService - Latest Migration Version [1.7.1]
fusionauth_1  | Jul 28, 2019 10:44:41.734 AM INFO  com.inversoft.maintenance.search.ElasticsearchMaintenanceModeSearchService - Connecting to Search Engine at [http://search:9200]
fusionauth_1  | Jul 28, 2019 10:44:42.797 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - Loading FusionAuth configuration file [/usr/local/fusionauth/fusionauth-app/apache-tomcat/../../config/fusionauth.properties]
fusionauth_1  | Jul 28, 2019 10:44:42.798 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - FusionAuth Runtime Mode [Production]
fusionauth_1  | Jul 28, 2019 10:44:42.798 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - Override property [fusionauth-app.url] set to [http://fusionauth:9011] using FUSIONAUTH_URL environment value.
fusionauth_1  | Jul 28, 2019 10:44:42.867 AM INFO  com.inversoft.maintenance.db.JDBCMaintenanceModeDatabaseService - Database Version [1.7.2]
db_1          | LOG:  incomplete startup packet
fusionauth_1  | Jul 28, 2019 10:44:42.867 AM INFO  com.inversoft.maintenance.db.JDBCMaintenanceModeDatabaseService - Latest Migration Version [1.7.1]
fusionauth_1  | Jul 28, 2019 10:44:42.868 AM INFO  io.fusionauth.app.maintenance.FusionAuthMaintenanceModeWorkflow - Determine database status : ACTIVE
fusionauth_1  | Jul 28, 2019 10:44:42.872 AM INFO  com.inversoft.maintenance.search.ElasticsearchMaintenanceModeSearchService - Connecting to Search Engine at [http://search:9200]
fusionauth_1  | Jul 28, 2019 10:44:42.996 AM INFO  io.fusionauth.app.maintenance.FusionAuthMaintenanceModeWorkflow - Determine search status : ACTIVE]
fusionauth_1  | Jul 28, 2019 10:44:42.998 AM INFO  org.primeframework.mvc.servlet.PrimeServletContextListener - Initializing Prime
fusionauth_1  | Jul 28, 2019 10:44:43.028 AM INFO  io.fusionauth.app.guice.FusionAuthModule - 
fusionauth_1  | 
fusionauth_1  | ---------------------------------------------------------------------------------------------------------
fusionauth_1  | ---------------------------------- Starting FusionAuth version [1.7.2] ----------------------------------
fusionauth_1  | ---------------------------------------------------------------------------------------------------------
fusionauth_1  | 
fusionauth_1  | Jul 28, 2019 10:44:43.652 AM INFO  io.fusionauth.api.plugin.guice.PluginModule - No plugins found
fusionauth_1  | Jul 28, 2019 10:44:44.155 AM INFO  io.fusionauth.api.service.system.NodeService - Starting Master node with Id [6e061ebe-104f-474f-9a51-a12915ce13c5]
fusionauth_1  | Jul 28, 2019 10:44:45.444 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - Loading FusionAuth configuration file [/usr/local/fusionauth/fusionauth-app/apache-tomcat/../../config/fusionauth.properties]
fusionauth_1  | Jul 28, 2019 10:44:45.445 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - FusionAuth Runtime Mode [Production]
fusionauth_1  | Jul 28, 2019 10:44:45.445 AM INFO  io.fusionauth.api.configuration.PropertiesFileFusionAuthConfiguration - Override property [fusionauth-app.url] set to [http://fusionauth:9011] using FUSIONAUTH_URL environment value.
fusionauth_1  | Jul 28, 2019 10:44:45.451 AM INFO  com.inversoft.jdbc.hikari.DataSourceProvider - Connecting to PostgreSQL database at [jdbc:postgresql://db:5432/fusionauth]
fusionauth_1  | Jul 28, 2019 10:44:45.476 AM INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
fusionauth_1  | Jul 28, 2019 10:44:45.537 AM INFO  com.zaxxer.hikari.pool.PoolBase - HikariPool-1 - Driver does not support get/set network timeout for connections. (Method org.postgresql.jdbc.PgConnection.getNetworkTimeout() is not yet implemented.)
fusionauth_1  | Jul 28, 2019 10:44:45.613 AM INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
fusionauth_1  | Jul 28, 2019 10:44:48.403 AM INFO  com.inversoft.scheduler.DefaultScheduler - Starting up scheduler
fusionauth_1  | Jul 28, 2019 10:44:48.405 AM INFO  com.inversoft.scheduler.DefaultScheduler - Scheduler is running
fusionauth_1  | Jul 28, 2019 10:44:48.433 AM INFO  io.fusionauth.api.service.system.NodeService - Add node with address [http://fusionauth:9011]
fusionauth_1  | Jul 28, 2019 10:44:48.794 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_2_0]
fusionauth_1  | Jul 28, 2019 10:44:48.802 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_3_0]
fusionauth_1  | Jul 28, 2019 10:44:48.810 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_3_1]
fusionauth_1  | Jul 28, 2019 10:44:48.816 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_5_0]
fusionauth_1  | Jul 28, 2019 10:44:48.829 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_6_0]
fusionauth_1  | Jul 28, 2019 10:44:48.838 AM INFO  com.inversoft.migration.Migrator - Run Cleanup [io.fusionauth.api.migration.guice.Migration_1_7_0]
fusionauth_1  | 28-Jul-2019 10:44:48.957 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-9011"]
fusionauth_1  | 28-Jul-2019 10:44:49.016 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["https-jsse-nio-9013"]
fusionauth_1  | 28-Jul-2019 10:44:49.032 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["ajp-nio-9019"]
fusionauth_1  | 28-Jul-2019 10:44:49.043 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in 18179 ms
fusionauth_1  | Jul 28, 2019 10:44:54.194 AM INFO  io.fusionauth.api.service.search.ElasticSearchClientProvider - Connecting to FusionAuth Search Engine at [http://search:9200]

```

**Comment by pitcons** [comment]
So in my case, the problem was in "blocks.read_only" of ElasticSearch.
We disable read_only mode then Fusionauth work was restored.

**Comment by nadilas** [comment]
I didn’t find any solution. Ended up redeploying fusionauth from scratch.

**Comment by robotdan** [comment]
Possible solution provided by @colundrum in Issue #370

**Comment by radicaljohan** [comment]
I resolved the issue by reindexing.
In the application menu go to:
FusionAuth -> System -> Reindex

**Comment by colundrum** [comment]
You was lucky because it's impossible to login for me

**Comment by MattCarr147** [comment]
I may have encountered a similar issue during initial setup this evening. Dunno if the following will be relevant, but just in case..

I'm running on Windows and installed and ran authfusion using `startup.bat` as recommended in the 5-minute guide docs, using postgres as the db. I hit the same error as described in the opening post here, only at the point of completing setup (when you add an admin account). 

My 'fusionauth-search' prompt was full of warnings indicating low disk space (in my case less than 5% was free). I can't remember the exact terminology and deleted the logs when trying to fix it (doh), but the error was something along the lines of the following:

`flood stage disk watermark 95 exceeded on ...`

I'm also sure it also mentioned that _something_ (sorry!) had been set to read-only, which I suspect is causing the error with authfusion. 

I freed up some space and this was reflected in the fusionauth-search prompt, however I still got the same error in the fusion instance. After that, I deleted the fusionauth folder entirely and reinstalled, connected to the previous database and this time it let me complete setup with no issues.
